### ISSUE TITLE
[Ask] Can View Another People Paste

### ISSUE THREAD

**Issue by secghost** [jordansamuel/PASTE#181]
Can you fix this parameter php?id= like this to private paste without login like [https://hatebin.com/ryypygzofi](https://hatebin.com/**ryypygzofi**) cause can raw a paste other people just change number on parameter like php?id=1 to php?id=2

**Comment by thinkverse** [comment]
Private pastes are private and should not be visible by changing the query parameters value. Even if a user manually adds a specific id to the query parameter, if it's set to private, it's private. It doesn't matter if they specify that they want either of the `raw|download|embed` versions either.

If that is was you're asking, I might be misunderstanding your question, let me know. 👍

![Screen Shot 2021-09-12 at 12 12 09 PM](https://user-images.githubusercontent.com/2221746/132983703-e5255a40-9664-4484-8a23-8fe654002bf3.png)

![Screen Shot 2021-09-12 at 12 12 24 PM](https://user-images.githubusercontent.com/2221746/132983702-0a2bc7dc-8f45-4ac8-a311-7f8a69e996fe.png)

**Comment by eviltwin-id** [comment]
this is need a login & register bro,and thats parameter make a people can view a paste just change 1 to 2 , like php?id=1 to php?id=2,can you change parameter this to random text like  https://hatebin.com/{ryypygzofi}?

**Comment by BGHDDevelopment** [comment]
@eviltwin-id Can you provide an example of what you mean in a practical environment? Like, provide example links or screenshots?

**Comment by thinkverse** [comment]
> this is need a login & register bro,and thats parameter make a people can view a paste just change 1 to 2 , like php?id=1 to php?id=2,can you change parameter this to random text like [https://hatebin.com/{ryypygzofi}](https://hatebin.com/%7Bryypygzofi%7D)?

Yes, I can agree that having the `new paste` button showing doesn't really make sense in this context and that a login/register button would be more appropriate.

As for changing the parameters from IDs to something else, that has been suggested before (https://github.com/jordansamuel/PASTE/issues/38) and will be added later. For right now I'm mostly interested in fixing issues and making sure paste will work on PHP 8+. New **large** features will be on the back burner until then.

**Comment by secghost** [comment]
@thinkverse thanks bro, i will wait it a ne feature to random URL ID , i just want it , thanks
